archive_viewer/av_file_convert.py

```diff
@@ -0,0 +1,374 @@
+            The data in a dictionary format
+        """
+        data_dict = {'connection_parameter': '',
+                    'plot_title': '',
+                    'legend_configuration': {},
+                    'time_axis': [],
+                    'range_axis': [],
+                    'pv': [],
+                    'formula': []}
+
+        data_dict['connection_parameter'] = xml.find("connection_parameter").text
+        data_dict['plot_title'] = xml.find("plot_title").text
+        data_dict['legend_configuration'] = xml.find("legend_configuration").attrib
+
+        for key in ("time_axis", "range_axis", "pv", "formula"):
+            for element in xml.findall(key):
+                ele_dict = element.attrib
+                ele_dict |= {sub_ele.tag: sub_ele.text for sub_ele in element}
+                data_dict[key].append(ele_dict)
+
+        return data_dict
+
+    @staticmethod
+    def get_plot_data(plot: PyDMTimePlot) -> Dict:
+        """Extract plot, axis, and curve data from a PyDMTimePlot object"""
+        output_dict = {'archiver_url': getenv("PYDM_ARCHIVER_URL"),
+                       'plot': {},
+                       'time_axis': {},
+                       'y-axes': [],
+                       'curves': []}
+
+        [start_ts, end_ts] = plot.getXAxis().range
+        start_dt = datetime.fromtimestamp(start_ts)
+        end_dt = datetime.fromtimestamp(end_ts)
+
+        output_dict['time_axis'] = {'name': "Main Time Axis",
+                                    'start': start_dt.isoformat(sep=' ', timespec='seconds'),
+                                    'end': end_dt.isoformat(sep=' ', timespec='seconds'),
+                                    'location': "bottom"}
+
+        for a in plot.getYAxes():
+            axis_dict = json.loads(a, object_pairs_hook=OrderedDict)
+            output_dict['y-axes'].append(axis_dict)
+
+        for c in plot.getCurves():
+            curve_dict = json.loads(c, object_pairs_hook=OrderedDict)
+            if not curve_dict['channel']:
+                continue
+            output_dict['curves'].append(curve_dict)
+
+        return output_dict
+
+    @staticmethod
+    def srgb_to_qColor(srgb: str) -> QColor:
+        """Convert RGB strings to QColors. The string is a 32-bit
+        integer containing the aRGB values of a color. (e.g. #FF0000 or -65536)
+
+        Parameters
+        ----------
+        srgb : str
+            Either a hex value or a string containing a signed 32-bit integer
+
+        Returns
+        -------
+        QColor
+            A QColor object storing the color described in the string
+        """
+        if not srgb:
+            return QColor()
+        elif srgb[0] != '#':
+            rgb_int = int(srgb) & 0xFFFFFFFF
+            srgb = f"#{rgb_int:08X}"
+        return QColor(srgb)
+
+    @staticmethod
+    def remove_null_values(dict_in: Dict) -> Dict:
+        """Remove all key-value pairs from a given dictionary where the value is None"""
```

Comment:
@YektaY do you want me to add the parameters & returns to this docstring?
It feels like a pretty straightforward helper function, but I can add more to the docs to match styling

Reply:
adding more would be helpful for the auto documentation we will have for the website

Reply:
Solved in commit 3f00fff13322325cd1f81101ea9b124ff61f02fb